lib/pipelines.js

```diff
@@ -317,289 +370,356 @@ async function http_delete_pipeline_coupling(pg_pool, req, res, regex) {
-      if(src_addons[i].addon_service.id === dst_addons[j].addon_service.id) {
+    for (let j = 0; j < dst_addons.length; j++) {
+      if (src_addons[i].addon_service.id === dst_addons[j].addon_service.id) {
         found = true;
       }
     }
-    if(!found) {
+    if (!found) {
       warnings_addon_source.push(src_addons[i]);
     }
   }
-  let warnings_addon_dest = [];
-  for (let i=0; i < dst_addons.length; i++) {
+  const warnings_addon_dest = [];
+  for (let i = 0; i < dst_addons.length; i++) {
     let found = false;
-    for (let j=0; j < src_addons.length; j++) {
-      if(dst_addons[i].addon_service.id === src_addons[j].addon_service.id) {
+    for (let j = 0; j < src_addons.length; j++) {
+      if (dst_addons[i].addon_service.id === src_addons[j].addon_service.id) {
         found = true;
       }
     }
-    if(!found) {
+    if (!found) {
       warnings_addon_dest.push(dst_addons[i]);
     }
   }
-  if(warnings_addon_source.length > 0) {
+  if (warnings_addon_source.length > 0) {
     warnings.push(`  The source app has addon(s) (${warnings_addon_source.map((x) => x.addon_service.name).join(', ')}) that are not on the destination app (${app_dst.name}-${app_dst.space})`);
   }
 
-  if(warnings_addon_dest.length > 0) {
+  if (warnings_addon_dest.length > 0) {
     warnings.push(`  The destination app has addon(s) (${warnings_addon_dest.map((x) => x.addon_service.name).join(', ')}) that are not on the source app (${app_dst.name}-${app_dst.space})`);
   }
 
-  return warnings
+  return warnings;
 }
 
-const create_pipeline_promotion_query = query.bind(query, fs.readFileSync('./sql/insert_pipeline_promotion.sql').toString('utf8'), (d) => { return d; });
-const get_pipeline_couplings_by_stage = query.bind(query, fs.readFileSync('./sql/select_pipeline_couplings_by_stage.sql').toString('utf8'), (d) => { return d; });
-const create_pipeline_promotion_target_query = query.bind(query, fs.readFileSync('./sql/insert_pipeline_promotion_target.sql').toString('utf8'), (d) => { return d; });
-const select_release_statuses = query.bind(query, fs.readFileSync('./sql/select_release_statuses.sql').toString('utf8'), (d) => { return d; });
-async function http_create_pipeline_promotion(pg_pool, req, res, regex) {
-  let payload = await httph.buffer_json(req)
+const create_pipeline_promotion_query = query.bind(query, fs.readFileSync('./sql/insert_pipeline_promotion.sql').toString('utf8'), (d) => d);
+const get_pipeline_couplings_by_stage = query.bind(query, fs.readFileSync('./sql/select_pipeline_couplings_by_stage.sql').toString('utf8'), (d) => d);
+const create_pipeline_promotion_target_query = query.bind(query, fs.readFileSync('./sql/insert_pipeline_promotion_target.sql').toString('utf8'), (d) => d);
+const select_release_statuses = query.bind(query, fs.readFileSync('./sql/select_release_statuses.sql').toString('utf8'), (d) => d);
+async function http_create_pipeline_promotion(pg_pool, req, res /* regex */) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Async function 'http_create_pipeline_promotion' has a complexity of 17. Maximum allowed is 8.](https://app.codacy.com/manual/Akkeris/controller-api/pullRequest?prid=5244943)